pulsar-client-go/examples/reader/reader.go

```diff
@@ -0,0 +1,48 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package main
+
+import (
+	"../../pulsar"
+	"fmt"
+	"log"
+)
+
+func main() {
+	client := pulsar.NewClient().
+		ServiceUrl("pulsar://localhost:6650").Build()
+
+	reader, err := client.NewReader().Topic("my-topic").StartFromEarliest().Create()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for {
+		msg, err := reader.ReadNext()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		fmt.Printf("Received message  msgId: %s -- content: '%s'\n",
+			msg.MessageId(), string(msg.Payload()))
+	}
```

Comment:
Should a reader ack a message?

Reply:
No, the reader needs to be managing the "message-id" manually, outside the scope of Pulsar, so no acks are needed

Reply:
Ok I thought so actually for the binary protocol.. I got confused because the Websocket docs requires the reader to ack (whether that is actually required or not on the server, I am not sure).